### PR TITLE
[INFRA][TEST] Add tests for torch.nn.functional.dropout

### LINKED ISSUE #318: Enable torch.nn.Dropout
### What feature would you like to be added?

Enable [torch.nn.Dropout](https://docs.pytorch.org/docs/stable/generated/torch.nn.Dropout.html)

### Why is this feature needed?

Used in important models.

### Any alternative solution that you may want to share?

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by imaihal**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

torch.nn.functional.dropout is already enabled. It is replaced with torch.clone(). This PR just add tests for torch.nn.functional.dropout.

#### Which issue(s) this PR is related to:

Fixes #318 

#### Special notes for your reviewer:

- Test results
```
$ pytest 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.12.9, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/imaihal/dt-inductor-4/torch-spyre
configfile: pyproject.toml
plugins: hypothesis-6.148.7
collected 206 items                                                                                                                                                                          

tests/_inductor/test_building_blocks.py .s.s..                                                                                                                                         [  2%]
tests/_inductor/test_inductor_ops.py ..s..........................................................................................................                                     [ 55%]
tests/tensor/test_tensor_layout.py ........                                                                                                                                            [ 59%]
tests/test_fallbacks.py ........                                                                                                                                                       [ 63%]
tests/test_modules.py ssssss.                                                                                                                                                          [ 66%]
tests/test_ops.py ..ssssss...................sss..ssss.s.......ss.ss                                                                                                                   [ 91%]
tests/test_spyre.py .s...s...........                                                                                                                                                  [ 99%]
tests/test_spyre_lazy_silent.py .                                                                                                                                                      [100%]

====================================================================================== warnings summary ======================================================================================
tests/_inductor/test_inductor_ops.py::TestOps::test_activation_cls_gelu_fp16
  /home/imaihal/dt-inductor-4/pytorch/torch/_inductor/ops_handler.py:745: UserWarning: undefined OpHandler.gelu, please add missing op schema
    warnings.warn(f"undefined OpHandler.{name}, please add missing op schema")

tests/_inductor/test_inductor_ops.py::TestOps::test_pointwise_range_op_clamp_fp16
  /home/imaihal/dt-inductor-4/pytorch/torch/_inductor/ops_handler.py:745: UserWarning: undefined OpHandler.clamp, please add missing op schema
    warnings.warn(f"undefined OpHandler.{name}, please add missing op schema")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================== 177 passed, 29 skipped, 2 warnings in 73.99s (0:01:13) ===================================================================
```

### CODE REVIEW

**Review by dgrove-oss — APPROVED** [review]
lgtm
